Dockerfile-base

```diff
@@ -0,0 +1,23 @@
+FROM python:3.12-slim
+WORKDIR /usr/src/app
+
+# set environment variables
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+
+# Install pip and dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    git \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install dependencies
+RUN pip install git+https://github.com/AI4quantum/maestro.git@v0.2.0
```

Comment:
is there a way to make the version tag customizable at build so we don't need to update this file every release? like it can default to `main` unless a tag is specified?

Reply:
or would it make sense to do a dev install here from the current directory since in theory this would be built from the root of the repo

Reply:
Also once this is merged it would be worth creating a followup task to publish it to ghcr as part of the release CI